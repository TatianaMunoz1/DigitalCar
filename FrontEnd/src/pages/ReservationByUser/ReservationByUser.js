import React, { useState, useEffect } from "react";
import Subheader from "../../components/Subheader/Subheader";
import styles from "./ReservationByUser.module.css";
import axios from "axios";
import Url from "../../Url";

function ReservationByUser(props) {
  const [reservations, setReservations] = useState([]);

  useEffect(() => {
    const params = {
      params: {
        userId: props.userId,
      },
    };
    axios
      .get(Url+"/booking/find", params)
      .then((response) => {
        if (response.data.length > 0) setReservations(response.data);
      })
      .catch((error) => {});
  }, []);

  const stateReservation = (index) => {
    if (new Date() < new Date(reservations[index].startDateTime.slice(0, 10))) {
      return "Pendiente";
    } else {
      return "Concretado";
    }
  };

  const reservationList = () =>
    reservations.map((reservation, index) => (
      <tr>
        <td>{index + 1}</td>
        <td>{reservation.startDateTime.slice(0, 10)}</td>
        <td>{reservation.endDate}</td>
        <td>{reservation.startDateTime.slice(11, 16)} hs</td>
        <td>{stateReservation(index)}</td>
      </tr>
    ));
  const reservationListMobile = () =>
    reservations.map((reservation, index) => (
      <tr>
        <tr>
          <td>PEDIDO # </td>
          <td> {index + 1}</td>
        </tr>
        <tr>
          <td>FECHA RETIRO </td>
          <td> {reservation.startDateTime.slice(0, 10)}</td>
        </tr>
        <tr>
          <td>FECHA DEVOLUCION </td>
          <td> {reservation.endDate}</td>
        </tr>
        <tr>
          <td>HORA RETIRO </td>
          <td> {reservation.startDateTime.slice(11, 16)} hs</td>
        </tr>
        <tr>
          <td>ESTADO </td>
          <td> {stateReservation(index)}</td>
        </tr>
      </tr>
    ));

  return (
    <div className={styles.container}>
      <div className={styles.container_header}>
        <Subheader title={"Mis Reservas"} link={"/"} />
      </div>
      <section className={styles.container_reservations}>
        <div>
          <div className={styles.container_table_mobile}>
            <h3>Lista de reservas</h3>
            <table>
              <tbody>{reservationListMobile()}</tbody>
            </table>
          </div>

          <table className={styles.container_table}>
            <thead className={styles.thead}>
              <tr>
                <th>PEDIDO #</th>
                <th>FECHA RETIRO</th>
                <th>FECHA DEVOLUCION</th>
                <th>HORA RETIRO</th>
                <th>ESTADO</th>
              </tr>
            </thead>
            <tbody>{reservationList()}</tbody>
          </table>
        </div>
      </section>
    </div>
  );
}

export default ReservationByUser;
