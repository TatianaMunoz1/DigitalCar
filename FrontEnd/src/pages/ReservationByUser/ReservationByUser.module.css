.container {
  background: #000000;
  position: absolute;
  top: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
}
.container_table_mobile {
  padding: 0px 15px;
}
h3 {
  line-height: 50px;
}
.container_reservations {
  color: #ffffff;
  width: 100%;
  height: 100%;
  padding-top: 200px;
  display: flex;
  justify-content: center;
}
.container_header {
  background: #000000;
  position: absolute;
  top: 0px;
  left: 0px;
  height: 90px;
  width: 100%;
}
.container_table {
  height: auto;
  text-align: center;
  display: none;
}
table {
  border-collapse: collapse;
}
tr {
  height: 45px;
}
tr:first-child {
  border-top: 1px solid #ffffff;
}
th {
  font-size: 14px;
  border-bottom: 1px solid #ffffff;
}
td {
  font-size: 14px;
}
.container_table_mobile td:first-child {
  padding: 0px 10px 0px 0px;
}

/* -----TABLET----- */
@media (min-width: 760px) and (max-width: 1280px) {
  .container_reservations {
    padding: 0px 30px;
    padding-top: 300px;
  }
  .container_table_mobile {
    display: none;
  }
  .container_table {
    display: contents;
    height: auto;
    text-align: center;
  }
  table {
    border-collapse: collapse;
  }

  tr {
    height: 50px;
    border-bottom: 1px solid #ffffff;
  }
  tr:last-child {
    border: none;
  }
  th {
    padding: 0px 30px;
    margin: 0px;
    font-size: 16px;
    border-bottom: 1px solid #ffffff;
  }
  td {
    font-size: 15px;
  }
}

/* -----DESKTOPS----- */
@media (min-width: 1281px) {
  .container_table_mobile {
    display: none;
  }
  .container_table {
    display: contents;
    height: auto;
    text-align: center;
  }
  .container_reservations {
    padding-top: 300px;
  }
  table {
    border-collapse: collapse;
  }
  tr {
    height: 50px;
    border-bottom: 1px solid #ffffff;
  }
  tr:last-child {
    border: none;
  }
  th {
    padding: 0px 30px;
    margin: 0px;
    font-size: 18px;
    border-bottom: 1px solid #ffffff;
  }
  td {
    font-size: 16px;
  }
}
