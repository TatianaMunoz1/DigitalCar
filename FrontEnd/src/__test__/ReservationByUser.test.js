import React from "react";
import { mount, shallow } from "enzyme";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import ReservationByUser from "../pages/ReservationByUser/ReservationByUser.js";

describe("ReservationByUser", () => {
  const history = createMemoryHistory();
  const dataReservations = [
    {
      id: 1,
      startDateTime: "2021-01-01T07:00:00.5",
      endDate: "2022-01-08",
    },
    {
      id: 2,
      startDateTime: "2020-01-01T07:00:00.5",
      endDate: "2022-01-08",
    },
    {
      id: 3,
      startDateTime: "2022-01-01T07:00:00.5",
      endDate: "2022-01-08",
    },
    {
      id: 4,
      startDateTime: "2022-01-01T22:00:00.5",
      endDate: "2022-01-08",
    },
  ];
  
  test("render Reservation by user ", () => {
        const wrapper = shallow(<ReservationByUser/>)
        expect(wrapper).toMatchSnapshot();
    });

  test("must be exist", () => {
    const wrapper = mount(
      <Router history={history}>
        <ReservationByUser {...dataReservations} />
      </Router>
    );
    expect(wrapper.text().includes("Mis Reservas")).toBeTruthy();
    expect(wrapper.text().includes("PEDIDO #")).toBeTruthy();
    expect(wrapper.text().includes("FECHA RETIRO")).toBeTruthy();
    expect(wrapper.text().includes("FECHA DEVOLUCION")).toBeTruthy();
    expect(wrapper.text().includes("HORA RETIRO")).toBeTruthy();
    expect(wrapper.text().includes("ESTADO")).toBeTruthy();
  });
});
