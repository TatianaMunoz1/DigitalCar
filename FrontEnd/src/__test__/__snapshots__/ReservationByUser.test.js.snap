// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationByUser render Reservation by user  1`] = `ShallowWrapper {}`;
